recursive.md

```diff
@@ -2,6 +2,108 @@
 
 ![broccoli](images/broccoli.png)
 
+## recursive computations
+
+This explanation is from [Big Java: Early Objects 7ed](https://www.wiley.com/en-us/Big+Java%3A+Early+Objects%2C+7th+Edition-p-9781119499091) by Cay Horstmann.
+
+A recrusive computation solves a problem by using the solution to a simpler version of the problem, to solve the original problem.
+
+For example, if we look at a triangle made from shapes such as this one:
+```
+[]
+[][]
+[][][]
+```
+We’d like to compute the area of the triangle assuming each square has area 1.
+
+You may know there is a simple formula for computing the area of a triangle, `base * height / 2`, but the purpose of this example is to explain the concept of recursion by walking through an example.
+
+If the width of the triangle is 1, then the triangle consists of a single square, and the area is 1.
+
+To deal with the general case, consider this picture:
+```
+[]
+[][]
+[][][]
+[][][][]
+```
```

Comment:
Let's add empty lines above and below the code block; also declare some language for it (doesn't matter what), like "```python". Otherwise Markdown Lint will complain.